src/js/actions/libraries.js

```diff
@@ -544,10 +552,18 @@ define(function (require, exports) {
      * @return {Promise}
      */
     var selectLibrary = function (id) {
-        return this.dispatchAsync(events.libraries.LIBRARY_SELECTED, { id: id });
+        var updatePreferencePromise = this.transfer(
+            preferencesActions.setPreference, _LAST_SELECTED_LIBRARY_ID_PREF, id);
+        
+        return Promise.resolve(updatePreferencePromise)
```

Comment:
I'd say dispatching the library selected event is higher priority than setting the preference here.